sem/utils.py

```diff
@@ -544,7 +544,15 @@ def parse_logs(log_file):
         log_file (string): Path to where the log file is stored
     """
     log_list = []
-    regex = re.compile(r'^[\+\-]?(\d+\.\d+)s ((?:\d+|-\d+)) (?:\[(.*?)\] )?(\w+):(\w+)\((.*?)\)(?:: \[(\w+)\s*\] (.*))?$')
+
+    # Regex for parsing the logs
+    time_re = r'[\+\-]?(?P<time>\d+\.\d+)s '
+    context_extended_context_re = r'(?P<context>(?:\d+|-\d+)) (?:\[(?P<extended_context>.*?)\] )?'
+    component_function_arguments_re = r'(?P<component>\w+):(?P<function>\w+)\((?P<arguments>.*?)\)'
+    severity_class_message_re = r'(?:: \[(?P<severity_class>\w+)\s*\] (?P<message>.*))?'
```

Comment:
i'm thinking about maintaining this code. right now it's clear why these regexps should work, but in a few months the sem maintainer(s) (probably davide) might not remember why some parts might not be there or why they are written in a specific way. it might be useful to add some comments for future maintainability

Reply:
Added in [7f32644](https://github.com/signetlabdei/sem/pull/51/commits/7f326449db9c0fb5ca76f4197a9dd236f5dcb0b3).